contracts/chain-adapters/Scroll_Adapter.sol

```diff
@@ -0,0 +1,126 @@
+import "./interfaces/AdapterInterface.sol";
+
+/**
+ * @title Scroll_Adapter
+ * @notice Adapter contract deployed on L1 alongside the HubPool to facilitate token transfers
+ * and arbitrary message relaying from L1 to L2.
+ */
+contract Scroll_Adapter is AdapterInterface {
+    using SafeERC20 for IERC20;
+
+    /**
+     * @notice Used as the gas limit for relaying messages to L2.
+     */
+    uint32 public constant l2GasLimit = 250_000;
+
+    /**
+     * @notice The address of the official l1GatewayRouter contract for Scroll for bridging tokens from L1 -> L2
+     * @dev We can find these (main/test)net deployments here: https://docs.scroll.io/en/developers/scroll-contracts/#scroll-contracts
+     */
+    IL1GatewayRouter public immutable l1GatewayRouter;
+
+    /**
+     * @notice The address of the official messenger contract for Scroll from L1 -> L2
+     * @dev We can find these (main/test)net deployments here: https://docs.scroll.io/en/developers/scroll-contracts/#scroll-contracts
+     */
+    IL1ScrollMessenger public immutable l1ScrollMessenger;
+
+    /**
+     * @notice The address of the official gas price oracle contract for Scroll for estimating the relayer fee
+     * @dev We can find these (main/test)net deployments here: https://docs.scroll.io/en/developers/scroll-contracts/#scroll-contracts
+     */
+    IL2GasPriceOracle public immutable l2GasPriceOracle;
+
+    /**************************************
+     *          PUBLIC FUNCTIONS          *
+     **************************************/
+
+    /**
+     * @notice Constructs new Adapter.
+     * @param _l1GatewayRouter Standard bridge contract.
+     * @param _l1ScrollMessenger Scroll Messenger contract.
+     * @param _l2GasPriceOracle Gas price oracle contract.
+     */
+    constructor(
+        IL1GatewayRouter _l1GatewayRouter,
+        IL1ScrollMessenger _l1ScrollMessenger,
+        IL2GasPriceOracle _l2GasPriceOracle
+    ) {
+        l1GatewayRouter = _l1GatewayRouter;
+        l1ScrollMessenger = _l1ScrollMessenger;
+        l2GasPriceOracle = _l2GasPriceOracle;
+    }
+
+    /**
+     * @notice Send message to `target` on Scroll.
+     * @dev This message is marked payable because relaying the message will require
+     * a fee that needs to be propagated to the Scroll Bridge. It will not send msg.value
+     * to the target contract on L2.
+     * @param target L2 address to send message to.
+     * @param message Message to send to `target`.
+     */
+    function relayMessage(address target, bytes calldata message) external payable {
+        // We can specifically send a message with 0 value to the Scroll Bridge
+        // and it will not forward any ETH to the target contract on L2. However,
+        // we need to set the payable value to msg.value to ensure that the Scroll
+        // Bridge has enough gas to forward the message to L2.
+        l1ScrollMessenger.sendMessage{ value: _generateRelayerFee() }(target, 0, message, l2GasLimit);
```

Comment:
If our gas limit is too high and gas is left unused on the destination, does that money get lost or is it refunded somehow?

Reply:
Per the TG group chat with the Scroll team, our ETH may get burned if we overestimate our gas limit. However, we were told in a previous message that 250,000 is an acceptable amount.